homeassistant/components/yamaha_musiccast/__init__.py

```diff
@@ -1 +1,171 @@
-"""The yamaha_musiccast component."""
+"""The MusicCast integration."""
+from __future__ import annotations
+
+import abc
+import asyncio
+from datetime import timedelta
+import logging
+
+from aiomusiccast.musiccast_device import MusicCastData, MusicCastDevice
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import service
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from ...helpers.entity import DeviceInfo, Entity
+from .const import BRAND, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the MusicCast component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up MusicCast from a config entry."""
+
+    client = MusicCastDevice(
+        entry.data[CONF_HOST], async_get_clientsession(hass), hass.loop
+    )
+    coordinator = MusicCastDataUpdateCoordinator(hass, client=client)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
```

Comment:
```suggestion
    await coordinator.async_config_entry_first_refresh()
```
https://github.com/home-assistant/core/blob/76edfe6c62c7ad49206813f2231ff79c4d99e2bb/homeassistant/helpers/update_coordinator.py#L149-L155